### PR TITLE
SLCORE-370 Update to sonar-plugin-api 9.5

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: analysis-engine/src/main/java/org/sonarsource/sonarlint/core/analysis/sonarapi/DefaultSensorContext.java [hunk 1]
```diff
@@ -188,4 +190,24 @@ private static UnsupportedOperationException unsupported() {
...
+  }
+
+  @Override
+  public boolean isCacheEnabled() {
+    return false;
+  }
+
+  @Override
+  public ReadCache previousCache() {
+    throw unsupported();
```
Comment (damien-urruty-sonarsource) [inline analysis-engine/src/main/java/org/sonarsource/sonarlint/core/analysis/sonarapi/DefaultSensorContext.java:205]: Will this be conditionally called on the analyzers side ? Else we could have a no-op read and write cache

File: pom.xml [hunk 2]
```diff
@@ -105,7 +106,13 @@
...
         <groupId>org.mockito</groupId>
         <artifactId>mockito-core</artifactId>
-        <version>4.4.0</version>
+        <version>${mockito.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.mockito</groupId>
+        <artifactId>mockito-junit-jupiter</artifactId>
+        <version>${mockito.version}</version>
+        <scope>test</scope>
```
Comment (damien-urruty-sonarsource) [inline pom.xml:115]: Could you explain why you had to add this ?

File: analysis-engine/pom.xml [hunk 3]
```diff
@@ -57,6 +57,11 @@
       <artifactId>mockito-core</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.mockito</groupId>
+      <artifactId>mockito-junit-jupiter</artifactId>
+      <scope>test</scope>
+    </dependency>
```
Comment (damien-urruty-sonarsource) [inline analysis-engine/pom.xml:64]: Same here

File: analysis-engine/src/main/java/org/sonarsource/sonarlint/core/analysis/sonarapi/DefaultSensorContext.java (hunk 1 above)
Comment (henryju) [reply analysis-engine/src/main/java/org/sonarsource/sonarlint/core/analysis/sonarapi/DefaultSensorContext.java:205]: Yeah, normally they should call it only when `isCacheEnabled` returns `true`

File: pom.xml (hunk 2 above)
Comment (henryju) [reply pom.xml:115]: I will extract the change in a separate commit to make it more obvious

**Review by damien-urruty-sonarsource — APPROVED** [review]
The <mockito.version> property should also be moved to this new commit then.
It's clearer with this split, thanks!

**Pull request merged**